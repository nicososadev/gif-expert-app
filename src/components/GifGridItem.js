import React from 'react'
import propTypes from 'prop-types'

export const GifGridItem = ( { title, url } ) => {
    
    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={ url } alt={ title } />
            <p>{ title }</p>
        </div>
    )
}

GifGridItem.propType = {
    title: propTypes.string.isRequired,
    url: propTypes.string.isRequired
}
