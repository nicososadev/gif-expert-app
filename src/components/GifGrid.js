import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import { GifGridItem } from './GifGridItem'
import { useFetchGifs } from '../hooks/useFetchGifs'

export const GifGrid = ({ category }) => {

    const { data:images , loading } = useFetchGifs( category )
    
    return (
        <Fragment>
            <h3>{ category }</h3>

            { loading && <p className="card animate__animated animate__flash">Loading...</p> }

            <div className="card-grid">
            
                {
                    images.map( img => (
                        <GifGridItem 
                            key={ img.id }
                            { ...img }
                        />
                    ))
                }
                
            </div>
        </Fragment>
    )
}

GifGrid.propTypes = {

    category: PropTypes.string.isRequired

}
