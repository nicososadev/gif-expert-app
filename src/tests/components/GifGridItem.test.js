import '@testing-library/jest-dom'
import React from 'react'
import { shallow } from 'enzyme'
import { GifGridItem } from '../../components/GifGridItem'

describe('Pruebas en el componente GifGridItem', () => {
    
    const title = 'Titulo'
    const url = 'https://url.com'

    let wrapper = shallow( <GifGridItem 
        title={ title }
        url={ url }
    />)

    test('Debe mostrar el componente correctamente', () => {

        expect( wrapper ).toMatchSnapshot()
    })
    
    test('Debe mostrar correctamente el titulo', () => {
        
        const p = wrapper.find('p').text().trim()

        expect( p ).toBe( title )

    })

    test('Debe mostrar correctamente la imagen', () => {
        
        const img = wrapper.find('img').props()

        expect( img.src ).toBe( url )
        expect( img.alt ).toBe( title )
        

    })

    test('La clase debe tener correctamente la animacion fadeIn', () => {
        
        const div = wrapper.find('div').props()

        expect( div.className ).toContain('animate__fadeIn')
    })
    
})
