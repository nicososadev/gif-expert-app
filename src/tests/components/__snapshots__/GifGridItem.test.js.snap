// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGridItem Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo"
    src="https://url.com"
  />
  <p>
    Titulo
  </p>
</div>
`;
